scipy/optimize/_minimize.py

```diff
@@ -808,9 +853,50 @@ def minimize_scalar(fun, bracket=None, bounds=None, args=(),
         raise ValueError('Unknown solver %s' % method)
 
 
+def _remove_from_bounds(bounds, i_fixed):
+    """Removes bounds for which upper and lower parts are equal"""
+    bounds.lb = bounds.lb[~i_fixed]
+    bounds.ub = bounds.ub[~i_fixed]
+    return bounds
+
+
+def _remove_from_func(fun_in, i_fixed, x_fixed, min_dim=None, remove=0):
+    """Wraps a function such that fixed variables need not be passed in"""
+    def fun_out(x_in, *args, **kwargs):
+        x_out = np.zeros_like(i_fixed, dtype=x_in.dtype)
+        x_out[i_fixed] = x_fixed
+        x_out[~i_fixed] = x_in
+        y_out = fun_in(x_out, *args, **kwargs)
+        y_out = np.array(y_out)
+
+        if min_dim == 1:
+            y_out = np.atleast_1d(y_out)
+        elif min_dim == 2:
+            y_out = np.atleast_2d(y_out)
+
+        if remove == 1:
+            y_out = y_out[..., ~i_fixed]
+        elif remove == 2:
+            y_out = y_out[~i_fixed, ~i_fixed]
```

Comment:
I think we can remove this.

Reply:
If it's obvious how to recreate it, then by all means delete. It might come in handy in the future if we end up doing more methods.

Reply:
OK, I'll leave these in.